core/src/main/scala/cats/data/Kleisli.scala

```diff
@@ -718,6 +726,13 @@ private trait KleisliDistributive[F[_], R] extends Distributive[Kleisli[F, R, *]
   def map[A, B](fa: Kleisli[F, R, A])(f: A => B): Kleisli[F, R, B] = fa.map(f)
 }
 
+private trait KleisliAlign[F[_], R] extends Align[Kleisli[F, R, *]] {
+  def FA: Align[F]
+
+  override def align[A, B](fa: Kleisli[F, R, A], fb: Kleisli[F, R, B]): Kleisli[F, R, Ior[A, B]] =
+    Kleisli(r => FA.align(fa.run(r), fb.run(r)))
```

Comment:
I think this is a stack-safety trap, but it's not obvious to me if there's a better way.